includes/class-wc-stripe-payment-tokens.php

```diff
@@ -160,6 +209,122 @@ public function woocommerce_get_customer_payment_tokens( $tokens, $customer_id,
 		return $tokens;
 	}
 
+	/**
+	 * Gets saved tokens from Intentions API if they don't already exist in WooCommerce.
+	 *
+	 * @param array  $tokens     Array of tokens
+	 * @param string $user_id    WC User ID
+	 * @param string $gateway_id WC Gateway ID
+	 *
+	 * @return array
+	 */
+	public function woocommerce_get_customer_upe_payment_tokens( $tokens, $user_id, $gateway_id ) {
+		if ( is_user_logged_in() ) {
+			if ( count( $tokens ) >= get_option( 'posts_per_page' ) ) {
+				// The tokens data store is not paginated and only the first "post_per_page" (defaults to 10) tokens are retrieved.
+				// Having 10 saved credit cards is considered an unsupported edge case, new ones that have been stored in Stripe won't be added.
+				return $tokens;
+			}
```

Comment:
If this isn't here, you get into a state below where this WC hook returns fewer tokens than are saved in Stripe and we continually keep re-saving these tokens as WC Payment Tokens, since the block below believes that there are tokens saved in Stripe that are missing in WC.